graphql_compiler/tests/schema_transformation_tests/test_rename_schema.py

```diff
@@ -0,0 +1,721 @@
+              id: String
+            }
+
+            type Human2 {
+              id: String
+            }
+
+            type SchemaQuery {
+              Human: Human
+              Human2: Human2
+            }
+        ''')
+
+        with self.assertRaises(SchemaNameConflictError):
+            rename_schema(parse(schema_string), {'Human2': 'Human'})
+
+    def test_clashing_type_one_unchanged_rename(self):
+        schema_string = dedent('''\
+            schema {
+              query: SchemaQuery
+            }
+
+            type Human {
+              id: String
+            }
+
+            type Human2 {
+              id: String
+            }
+
+            type SchemaQuery {
+              Human: Human
+              Human2: Human2
+            }
+        ''')
+
+        with self.assertRaises(SchemaNameConflictError):
+            rename_schema(parse(schema_string), {'Human': 'Human', 'Human2': 'Human'})
+
+    def test_clashing_scalar_type_rename(self):
+        schema_string = dedent('''\
+            schema {
+              query: SchemaQuery
+            }
+
+            type Human {
+              id: String
+            }
+
+            scalar SCALAR
+
+            type SchemaQuery {
+              Human: Human
+            }
+        ''')
+
+        with self.assertRaises(SchemaNameConflictError):
+            rename_schema(parse(schema_string), {'Human': 'SCALAR'})
+
+    def test_builtin_type_conflict_rename(self):
+        schema_string = dedent('''\
+            schema {
+              query: SchemaQuery
+            }
+
+            type Human {
+              id: String
+            }
+
+            type SchemaQuery {
+              Human: Human
+            }
+        ''')
+
+        with self.assertRaises(SchemaNameConflictError):
+            rename_schema(parse(schema_string), {'Human': 'String'})
+
+    def test_invalid_schema(self):
+        with self.assertRaises(SchemaStructureError):
+            rename_schema(parse(ISS.invalid_schema), {})
+
+    def test_schema_extension(self):
+        schema_string = dedent('''\
+            schema {
+              query: SchemaQuery
+            }
+
+            type SchemaQuery {
+              Human: Human
+            }
+
+            type Human {
+              id: String
+            }
+
+            extend type Human {
+              age: Int
+            }
+        ''')
+        with self.assertRaises(SchemaStructureError):
+            rename_schema(parse(schema_string), {})
+
+    def test_input_type_definition(self):
+        schema_string = dedent('''\
+            schema {
+              query: SchemaQuery
+            }
+
+            type SchemaQuery {
+              id: String
+            }
+
+            input MessageInput {
+              content: String
+            }
+        ''')
+        with self.assertRaises(SchemaStructureError):
+            rename_schema(parse(schema_string), {})
+
+    def test_mutation_definition(self):
```

Comment:
What do you think of splitting up schema validation and renaming since the validation seems so comprehensive? @obi1kenobi

Reply:
Would probably be a bit easier to work with that way, sure.

Reply:
So splitting up input schema validation tests and renaming tests into two files?

Reply:
Or splitting up the code?